addons/purchase_stock/wizard/product_replenish.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.osv.expression import AND
+
+
+class ProductReplenish(models.TransientModel):
+    _inherit = 'product.replenish'
+
+    supplier_id = fields.Many2one("product.supplierinfo", string="Vendor")
+    show_vendor = fields.Boolean(compute="_compute_show_vendor")
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        if res.get('product_id'):
+            product_id = self.env['product.product'].browse(res['product_id'])
+            product_tmpl_id = product_id.product_tmpl_id
+            if 'warehouse_id' not in res:
+                company = product_tmpl_id.company_id or self.env.company
+                res['warehouse_id'] = self.env['stock.warehouse'].search([('company_id', '=', company.id)], limit=1).id
+            orderpoint = self.env['stock.warehouse.orderpoint'].search([('product_id', '=', product_tmpl_id.product_variant_id.id), ("warehouse_id", "=", res['warehouse_id'])], limit=1)
```

Comment:
we need to search for an orderpoint on the product_id itself or any variant of the current template